domain/domain.go

```diff
@@ -142,43 +142,57 @@ func (do *Domain) tryReload() {
 	}
 }
 
+const minReloadTimeout = 20 * time.Second
+
 func (do *Domain) reload() error {
 	// lock here for only once at same time.
 	do.m.Lock()
 	defer do.m.Unlock()
 
-	err := kv.RunInNewTxn(do.store, false, do.loadInfoSchema)
-	if err != nil {
-		return errors.Trace(err)
+	timeout := do.ddl.GetLease() / 2
+	if timeout < minReloadTimeout {
+		timeout = minReloadTimeout
 	}
 
-	atomic.StoreInt64(&do.lastLeaseTS, time.Now().UnixNano())
-	return nil
+	done := make(chan error, 1)
+	go func() {
+		var err error
+
+		for {
+			err = kv.RunInNewTxn(do.store, false, do.loadInfoSchema)
+			// if err is db closed, we will return it directly, otherwise, we will
+			// check reloading again.
+			if terror.ErrorEqual(err, localstore.ErrDBClosed) {
+				break
+			}
+
+			if err != nil {
+				// TODO: use a backoff algorithm.
```

Comment:
I think using a error level message is better, because this error seems serious.